src/index.html

```diff
@@ -14,6 +14,68 @@
   </head>
   <body>
     <h1 class="logo">MA</h1>
-    <script src="scripts/main.js"></script>
+    <div id="notifications"></div>
+    <script>
+      function createNotification(message, type) {
+        const notification = document.createElement('div');
+        notification.className = type;
+        notification.dataset.qa = 'notification';
+        notification.textContent = message;
+        document.getElementById('notifications').appendChild(notification);
+      }
+
+      const firstPromise = new Promise((resolve, reject) => {
+        let clicked = false;
+        document.addEventListener('click', (event) => {
+          if (event.button === 0) {
+            clicked = true;
+            resolve('First promise was resolved');
+          }
+        });
+
+        setTimeout(() => {
+          if (!clicked) {
+            reject('First promise was rejected');
+          }
+        }, 3000);
+      });
+
+      const secondPromise = new Promise((resolve) => {
+        document.addEventListener('click', (event) => {
+          if (event.button === 0 || event.button === 2) {
+            resolve('Second promise was resolved');
+          }
+        });
+      });
+
+      let leftClicked = false;
+      let rightClicked = false;
+      const thirdPromise = new Promise((resolve) => {
+        document.addEventListener('click', (event) => {
+          if (event.button === 0) {
+            leftClicked = true;
+          }
+          if (event.button === 2) {
+            rightClicked = true;
+          }
+          if (leftClicked && rightClicked) {
+            resolve('Third promise was resolved');
+          }
+        });
```

Comment:
You have multiple promises that add event listeners to the document. This can be optimized by using a single event listener that manages the logic for all promises, which would be more efficient and maintainable.